js/core/utils/dom.js

```diff
@@ -93,7 +93,11 @@ export const contains = (container, element) => {
     if(!element) {
         return false;
     }
+    const shadowHost = element.getRootNode && element.getRootNode().host;
+    return isInContainer(container, element) || shadowHost && isInContainer(container, shadowHost);
+};
 
+const isInContainer = (container, element) => {
     if(domAdapter.isTextNode(element)) {
```

Comment:
If we think that all this conditions are necessary for the case when target is a shadow host, it's a good idea to add tests for this cases